From: Cole Robinson <crobinso@redhat.com>
Date: Tue, 16 Apr 2019 20:14:12 -0400
Subject: [PATCH] Revert "target/i386: kvm: add VMX migration blocker"

This reverts commit d98f26073bebddcd3da0ba1b86c3a34e840c0fb8.

As is, it rejects libvirt managedsave and virt-manager snapshots
for default installed VMs on intel hosts. Upstream Paolo says that
the missing kernel bits are already queued up, and until then it's
fine to revert this.

https://lists.gnu.org/archive/html/qemu-devel/2019-04/msg02326.html
---
 target/i386/kvm.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/target/i386/kvm.c b/target/i386/kvm.c
index 3b29ce5c0d..6ad450d9f1 100644
--- a/target/i386/kvm.c
+++ b/target/i386/kvm.c
@@ -906,7 +906,6 @@ static int hyperv_init_vcpu(X86CPU *cpu)
 }
 
 static Error *invtsc_mig_blocker;
-static Error *vmx_mig_blocker;
 
 #define KVM_MAX_CPUID_ENTRIES  100
 
@@ -1270,17 +1269,6 @@ int kvm_arch_init_vcpu(CPUState *cs)
                                   !!(c->ecx & CPUID_EXT_SMX);
     }
 
-    if ((env->features[FEAT_1_ECX] & CPUID_EXT_VMX) && !vmx_mig_blocker) {
-        error_setg(&vmx_mig_blocker,
-                   "Nested VMX virtualization does not support live migration yet");
-        r = migrate_add_blocker(vmx_mig_blocker, &local_err);
-        if (local_err) {
-            error_report_err(local_err);
-            error_free(vmx_mig_blocker);
-            return r;
-        }
-    }
-
     if (env->mcg_cap & MCG_LMCE_P) {
         has_msr_mcg_ext_ctl = has_msr_feature_control = true;
     }
